OpenRA.Mods.Common/Projectiles/Bullet.cs

```diff
@@ -195,8 +209,32 @@ public void Tick(World world)
 			if (info.ContrailLength > 0)
 				contrail.Update(pos);
 
+			var flightLengthReached = ticks++ >= length;
+			var shouldBounce = bounceCount > 0;
+
+			if (flightLengthReached && shouldBounce)
+			{
+				shouldExplode |= AnyValidTargetsInRadius(world, pos, info.Width + info.TargetExtraSearchRadius, args.SourceActor, true);
+				var dir = new WVec(0, -1024, 0).Rotate(WRot.FromFacing(facing));
```

Comment:
Is there any reason why you can't simplify this to `target += (pos - source) * info.BounceRangeModifier / 100` and drop both `distance` and `dir`?